### PR TITLE
MB-10181 : Major change to circleci build and integration tests with close to 2x speedup

### PR DESCRIPTION
**Pull request by ahobson**
## Description
* Extract openapi so server build does not depend on full yarn install
* Build client and server artifacts once and then persist artifacts to
  workspace so that future steps can use without needed to checkout
  source or build again
* Use v3 for circleci go mod sources because of path change
* Use circleci ramdisk for speed where appropriate and possible
* Run integration tests in separate steps by office/admin/my both for
  speed and so if an integration test is flaky, we don't have to re-run
  the entire suite.
* Split integration tests by filesize when running in parallel to
  hopefully balance out runtime.
* Add lots of comments to explain the approach
* Speedup in CI is close to 2x (PRs go from from taking 27m-30m to 15m-17m)
* Try including junit output for client tests

## Reviewer Notes

* Look at the dependency graph (probably in CircleCI) to make sure I didn't miss anything
* Run the cypress tests locally since that script changed

## Setup

* Look at CircleCI?
* I'll need to test this against experimental to ensure deploys work, and then we'll need to double check deploys to every environment
* `make e2e_test_docker` is how to test locally

## Code Review Verification Steps

* [x] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [x] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](tbd) for this change

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

**Review by felipe-lee — COMMENTED** [review]
Haven't had a chance to review this in detail yet, but just as a quick comment, it looks like the swagger files got changed directly in `swagger/` instead of `swagger-def/` which means the next time they get generated, these changes will get lost, so I think those should get moved. I'll try reviewing more in detail later today.

File: docker-compose.e2e.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      - MUTUAL_TLS_ENABLED=0
+      - NO_TLS_ENABLED=1
+      - NO_TLS_PORT=4000
+      - PGPASSWORD=mysecretpassword
+      - REDIS_HOST=redis
+      - SERVE_ADMIN=true
+      - SERVE_API_GHC=true
+      - SERVE_API_INTERNAL=true
+      - SERVE_API_PRIME=false
+      - SERVE_API_SUPPORT=false
```
Comment (duncan-truss) [inline docker-compose.e2e.yml:93]: We'll want to add `- SERVE_PRIME_SIMULATOR=true` here since we now have an e2e specific docker-compose and prime simulator Cypress tests.

I just added it to the parent docker-compose because there was no e2e file before and running the Cypress tests in docker locally used the docker-compose.local.yml before.  You should be able to remove it with your changes.
https://github.com/transcom/mymove/commit/d392789f59933e451153d9760871edad6ab3647b#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R105

File: Makefile [hunk 2]
```diff
@@ -455,24 +455,34 @@ server_test_coverage: db_test_reset db_test_migrate redis_reset server_test_cove
 
 .PHONY: redis_pull
 redis_pull: ## Pull redis image
+ifdef CIRCLECI
+	@echo "Relying on CircleCI to setup redis."
+else
 	docker pull $(REDIS_DOCKER_CONTAINER_IMAGE)
+endif
```
Comment (gidjin) [inline Makefile:463]: I appreciate this, I wonder if there is a better way to split up our local setup commands from what CircleCI needs vs all the ifs. Not really part of this PR just wondering.

File: .circleci/config.yml [hunk 3]
```diff
@@ -75,72 +79,81 @@ executors:
...
     resource_class: medium+
-    working_directory: ~/transcom/mymove
+    working_directory: /mnt/ramdisk
     docker:
       - image: *circleci-docker
         auth:
           username: $DOCKERHUB_USERNAME
           password: $DOCKERHUB_PASSWORD
-  mymove_large:
+  mymove_compiler:
```
Comment (gidjin) [inline .circleci/config.yml:106]: I like the name change, doubt anyone was looking at them really but it feels better to name it after the purpose.

File: .circleci/config.yml [hunk 4]
```diff
@@ -910,41 +903,16 @@ jobs:
...
-          at: /home/circleci/transcom/mymove/bin
-      - run:
-          name: Output processes and resource usage every 5 seconds while job is running
-          command: |
-            while true; do
-              sleep 5
-              ps auxwwf
-              echo "======"
-            done
-          background: true
```
Comment (gidjin) [inline .circleci/config.yml:938]: I wonder if this is something we should keep handy for debugging or maybe it's just easy enough to look up and add if we need it.

File: docker-compose.e2e.yml (hunk 1 above)
Comment (gidjin) [reply docker-compose.e2e.yml:93]: Can we read these from one of the `config/env` files instead of having to repeat a lot of them in the docker compose or make file? Not really part of the PR but might be a good optimization.

**Review by gidjin — COMMENTED** [review]
Are there directions on running the cypress tests locally? I seem to be getting failures running `make e2e_clean e2e_test_fresh e2e_test`

I did a `make build` then the above and that seems to have started the tests. But after building and trying to run I have changes in the generated files that are not checked in, those probably need to be committed

**Comment by ahobson** [comment]
> Haven't had a chance to review this in detail yet, but just as a quick comment, it looks like the swagger files got changed directly in `swagger/` instead of `swagger-def/` which means the next time they get generated, these changes will get lost, so I think those should get moved. I'll try reviewing more in detail later today.

So ... it seems when running via `npx` it changes the generated output, I think because it winds up pulling in a different version of the YAML serializer.

Are you seeing something different?

File: Makefile (hunk 2 above)
Comment (ahobson) [reply Makefile:463]: Yeah, I'd love to improve, but I felt like I had already put enough into this one PR

File: docker-compose.e2e.yml (hunk 1 above)
Comment (ahobson) [reply docker-compose.e2e.yml:93]: Yeah, I'd love to find a way to reduce the number of `Dockerfile`s and `docker-compose*.yml` files, but I wanted to do that in another PR. We have so much cleanup to do and I needed to draw the line somewhere

Comment (ahobson) [reply docker-compose.e2e.yml:93]: @duncan-truss ah, ok, thanks for the catch!

**Comment by ahobson** [comment]
> Are there directions on running the cypress tests locally? I seem to be getting failures running `make e2e_clean e2e_test_fresh e2e_test`
> 
> I did a `make build` then the above and that seems to have started the tests. But after building and trying to run I have changes in the generated files that are not checked in, those probably need to be committed

Turns out the `make e2e_test` runs completely different because it's running the server and cypress locally.

`make e2e_test_docker` is the relevant one here

File: .circleci/config.yml (hunk 3 above)
Comment (ahobson) [reply .circleci/config.yml:106]: It was really helpful to me

File: .circleci/config.yml (hunk 4 above)
Comment (ahobson) [reply .circleci/config.yml:938]: I'll confess I've never looked at it after any failure, so I'm not sure it's useful. I'll add it back if folks want it

Comment (gidjin) [reply .circleci/config.yml:938]: It was lifted from CircleCI docs as I recall, so if we run into performance issues we can get it again. As I recall I was trying to see if a job was running out of memory at the time.

**Comment by gidjin** [comment]
I pulled and had to run `make server_generate` to regenerate the server files for some reason. But after doing that I have changes in `pkg/assets/assets.go` which I wasn't expecting. Though the diff is unclear. But when I run `make server_run` the build fails? Maybe it's my setup. 

```sh
 database "dev_db" already exists
Migrating the dev_db database...
DB_DEBUG=0 bin/milmove migrate -p "file://migrations/app/secure;file://migrations/app/schema" -m "migrations/app/migrations_manifest.txt"
/bin/sh: bin/milmove: cannot execute binary file
make[1]: *** [db_dev_migrate_standalone] Error 126
```

```diff
0 ❯ git diff
diff --git a/pkg/assets/assets.go b/pkg/assets/assets.go
index c9e65cea93..6bf4b1ad41 100644
--- a/pkg/assets/assets.go
+++ b/pkg/assets/assets.go
@@ -101,7 +101,7 @@ func pkgPaperworkFormtemplatesForm1203templatePng() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/form1203template.png", size: 217605, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/form1203template.png", size: 217605, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0xc4, 0x2b, 0x83, 0x5d, 0xac, 0xc9, 0x6c, 0xe7, 0x64, 0xdd, 0x96, 0x46, 0x9, 0xc8, 0xb6, 0x38, 0x2e, 0x7e, 0x41, 0xd6, 0x24, 0xee, 0xc, 0x27, 0x94, 0xc6, 0xc6, 0x3f, 0x24, 0x62, 0x26, 0xa4}}
        return a, nil
 }
@@ -121,7 +121,7 @@ func pkgPaperworkFormtemplatesShipment_summary_worksheet_page1Png() (*asset, err
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/shipment_summary_worksheet_page1.png", size: 159260, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/shipment_summary_worksheet_page1.png", size: 159260, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0xc2, 0x4a, 0x36, 0x4c, 0x1, 0x4d, 0x90, 0x74, 0x45, 0x28, 0xb3, 0xf4, 0x22, 0xc8, 0x3e, 0xea, 0x5d, 0x6b, 0x6e, 0xfa, 0x14, 0xe1, 0x53, 0xde, 0x85, 0xbb, 0x3, 0x6d, 0xcc, 0x2, 0x3f, 0x2}}
        return a, nil
 }
@@ -141,7 +141,7 @@ func pkgPaperworkFormtemplatesShipment_summary_worksheet_page2Png() (*asset, err
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/shipment_summary_worksheet_page2.png", size: 214003, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/shipment_summary_worksheet_page2.png", size: 214003, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0xdd, 0xd2, 0xa0, 0xad, 0xd2, 0x31, 0xbb, 0x59, 0xe9, 0x64, 0xfd, 0x51, 0xf8, 0x30, 0x45, 0xc4, 0x6c, 0xcd, 0x81, 0x96, 0xde, 0x82, 0xec, 0x5a, 0x60, 0x67, 0x97, 0xd3, 0x2d, 0x62, 0xf8, 0x14}}
        return a, nil
 }
@@ -161,7 +161,7 @@ func pkgPaperworkFormtemplatesShipment_summary_worksheet_page3Png() (*asset, err
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/shipment_summary_worksheet_page3.png", size: 125403, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/paperwork/formtemplates/shipment_summary_worksheet_page3.png", size: 125403, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x38, 0xb4, 0xf8, 0x1d, 0xa1, 0x5c, 0xed, 0xe0, 0xb7, 0x1d, 0x36, 0xd3, 0xfa, 0xe4, 0x6d, 0xef, 0x52, 0x75, 0x62, 0x21, 0x80, 0x16, 0xe0, 0xc3, 0xd1, 0x3f, 0xd6, 0xa5, 0xd7, 0xd, 0x1e, 0xc8}}
        return a, nil
 }
@@ -181,7 +181,7 @@ func pkgNotificationsTemplatesMove_approved_templateHtml() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_approved_template.html", size: 924, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_approved_template.html", size: 924, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x18, 0x50, 0xf9, 0x14, 0x1c, 0x9c, 0xe5, 0x52, 0xd7, 0xb9, 0x3f, 0x27, 0xc8, 0xde, 0xcf, 0x96, 0xd5, 0x77, 0xf, 0x2f, 0x50, 0xf2, 0x8e, 0x6, 0x3a, 0x44, 0xcf, 0x4c, 0x50, 0xf7, 0x97, 0xfb}}
        return a, nil
 }
@@ -201,7 +201,7 @@ func pkgNotificationsTemplatesMove_approved_templateTxt() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_approved_template.txt", size: 694, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_approved_template.txt", size: 694, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x91, 0x49, 0x3e, 0x28, 0x24, 0xbc, 0xed, 0x86, 0xab, 0x59, 0xce, 0x9b, 0xc7, 0x4a, 0x71, 0xc0, 0x53, 0x3b, 0xf3, 0xbe, 0xe2, 0x14, 0x6d, 0xb8, 0x16, 0xd1, 0xb2, 0x10, 0x9f, 0xcf, 0x4c, 0x1b}}
        return a, nil
 }
@@ -221,7 +221,7 @@ func pkgNotificationsTemplatesMove_canceled_templateHtml() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_canceled_template.html", size: 415, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_canceled_template.html", size: 415, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0xc2, 0x7f, 0x9e, 0xf7, 0x1c, 0x8b, 0x7, 0xcf, 0xb5, 0xf5, 0xe3, 0xa8, 0xe0, 0xe4, 0x84, 0x97, 0x60, 0x49, 0x28, 0x9, 0xae, 0xe5, 0xf8, 0xbb, 0xce, 0xe4, 0x22, 0x50, 0x5a, 0xed, 0x30, 0x71}}
        return a, nil
 }
@@ -241,7 +241,7 @@ func pkgNotificationsTemplatesMove_canceled_templateTxt() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_canceled_template.txt", size: 387, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_canceled_template.txt", size: 387, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x6f, 0x7, 0x88, 0x8d, 0xda, 0x3f, 0xa9, 0x30, 0xdb, 0x28, 0xf4, 0xb8, 0x9b, 0x4, 0xc2, 0x25, 0xc7, 0x1e, 0xa1, 0xbb, 0xc4, 0x16, 0xd0, 0x72, 0x2, 0x8f, 0xc2, 0x4, 0x5d, 0x44, 0x3f, 0x3}}
        return a, nil
 }
@@ -261,7 +261,7 @@ func pkgNotificationsTemplatesMove_payment_reminder_templateHtml() (*asset, erro
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_payment_reminder_template.html", size: 1741, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_payment_reminder_template.html", size: 1741, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x4f, 0xcf, 0xc6, 0x50, 0xb2, 0x2a, 0x71, 0xa5, 0x59, 0x32, 0x2b, 0xe5, 0x6c, 0xb2, 0xfe, 0xd7, 0xb7, 0x89, 0x1a, 0x7c, 0x79, 0x73, 0x81, 0x34, 0xf4, 0x8f, 0x60, 0xdc, 0x7d, 0x47, 0x2c, 0x6e}}
        return a, nil
 }
@@ -281,7 +281,7 @@ func pkgNotificationsTemplatesMove_payment_reminder_templateTxt() (*asset, error
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_payment_reminder_template.txt", size: 1546, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_payment_reminder_template.txt", size: 1546, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x69, 0xd9, 0xa9, 0xef, 0x9, 0x72, 0xd4, 0x4c, 0xa0, 0xea, 0x49, 0x71, 0xd2, 0xe8, 0x2, 0x0, 0xfb, 0x70, 0xd2, 0x2f, 0x66, 0xa, 0x84, 0x89, 0xf, 0xb, 0x74, 0x3a, 0xc4, 0xcc, 0x67, 0x1d}}
        return a, nil
 }
@@ -301,7 +301,7 @@ func pkgNotificationsTemplatesMove_reviewed_templateHtml() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_reviewed_template.html", size: 439, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_reviewed_template.html", size: 439, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x81, 0x33, 0x52, 0x69, 0x30, 0x15, 0xdb, 0x35, 0x59, 0x42, 0x79, 0x64, 0xe, 0xc9, 0x59, 0xea, 0x71, 0x54, 0x24, 0x13, 0xab, 0xf3, 0x31, 0xe8, 0xd1, 0xbb, 0xd7, 0x68, 0xd3, 0x71, 0xc3, 0xf3}}
        return a, nil
 }
@@ -321,7 +321,7 @@ func pkgNotificationsTemplatesMove_reviewed_templateTxt() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_reviewed_template.txt", size: 365, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_reviewed_template.txt", size: 365, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x89, 0x4e, 0xdb, 0x7, 0x11, 0xeb, 0x49, 0x86, 0x93, 0x7a, 0x1d, 0xe8, 0x6c, 0x60, 0xd2, 0x94, 0x41, 0xe5, 0xb, 0x1e, 0xb5, 0x5f, 0xc2, 0x74, 0x2e, 0x9b, 0xac, 0xf7, 0xc1, 0x14, 0x97, 0x2f}}
        return a, nil
 }
@@ -341,7 +341,7 @@ func pkgNotificationsTemplatesMove_submitted_templateHtml() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_submitted_template.html", size: 3382, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_submitted_template.html", size: 3382, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0xd2, 0x86, 0xb4, 0x1, 0x78, 0xd1, 0xe8, 0xf8, 0xda, 0xdd, 0x28, 0xd6, 0x38, 0x6b, 0x1c, 0x5f, 0xb4, 0x9e, 0xea, 0xaa, 0x37, 0x30, 0x7a, 0x8c, 0x98, 0x2d, 0xec, 0x7, 0x94, 0x88, 0x5, 0xcf}}
        return a, nil
 }
@@ -361,7 +361,7 @@ func pkgNotificationsTemplatesMove_submitted_templateTxt() (*asset, error) {
                return nil, err
        }

-       info := bindataFileInfo{name: "pkg/notifications/templates/move_submitted_template.txt", size: 2908, mode: os.FileMode(0644), modTime: time.Unix(1569961560, 0)}
+       info := bindataFileInfo{name: "pkg/notifications/templates/move_submitted_template.txt", size: 2908, mode: os.FileMode(0664), modTime: time.Unix(1569961560, 0)}
        a := &asset{bytes: bytes, info: info, digest: [32]uint8{0x22, 0x14, 0xe5, 0xed, 0xe, 0x30, 0x66, 0xe2, 0x60, 0x36, 0x79, 0x2e, 0x7, 0x61, 0xf6, 0x84, 0xd0, 0x86, 0x30, 0x46, 0x4c, 0xed, 0x5f, 0x7d, 0xd0, 0x15, 0xee, 0x62, 0xcf, 0xb7, 0x79, 0xb6}}
        return a, nil
 }
```

**Comment by ahobson** [comment]
> I pulled and had to run `make server_generate` to regenerate the server files for some reason. But after doing that I have changes in `pkg/assets/assets.go` which I wasn't expecting. Though the diff is unclear. But when I run `make server_run` the build fails? Maybe it's my setup.

I made some improvements to the scripts. It looks like you were caught with the binaries from linux on your Mac (or vice versa).

Try `make e2e_test_docker` with the latest code.

File: docker-compose.e2e.yml (hunk 1 above)
Comment (duncan-truss) [reply docker-compose.e2e.yml:93]: It looks like you set the serve prime api flags instead of the prime simulator, was that intentional or you are just going to keep the changes in docker-compose.yml that are in master?  If you merge in master you'll know if the Cypress tests for the Prime Simulator are working or failing.

**Comment by gidjin** [comment]
Okay I pulled, though I think I had the latest and ran `make clean`, then `make build` and finally `make e2e_test_docker` and everything ran and passed. I think I'm good to sign off on this, pending successful deploys to exp, and load test at least if not also demo.

**Comment by ahobson** [comment]
> Okay I pulled, though I think I had the latest and ran `make clean`, then `make build` and finally `make e2e_test_docker` and everything ran and passed. I think I'm good to sign off on this, pending successful deploys to exp, and load test at least if not also demo.

I tested a deployment to experimental and it uncovered a bug or two, but I've already updated this PR.

File: docker-compose.e2e.yml (hunk 1 above)
Comment (ahobson) [reply docker-compose.e2e.yml:93]: @duncan-truss Thank you! I've rebased on master and fixed the settings now.

**Review by gidjin — APPROVED** [review]
Changes worked locally and on exp. Approving
